tools/generate_features.py

```diff
@@ -345,16 +343,20 @@ def generate_features(
         catalog=gaia_catalog,
         query_radius_arcsec=bright_star_query_radius_arcsec,
         xmatch_radius_arcsec=xmatch_radius_arcsec,
-        limit=limit,
-        Ncore=Ncore,
     )
 
     print('Getting lightcurves...')
+    # For small source lists, shrink LC query limit until batching occurs
+    lc_limit = limit
+    if Ncore > 1:
+        while lc_limit > len(feature_gen_source_list):
+            lc_limit //= 10
```

Comment:
Shouldnt this be something like lc_limit  = np.floor(len(feature_gen_source_list) / Ncore) or something? Or is the // 10 still the best thing?